Add upload file action to immich integration (#39668)
home-assistant/home-assistant.github.io: source/_integrations/immich.markdown

### API key permissions

> For full functionality, enable the `album.read` and the `asset.upload` permission when creating your API key. Without this permission, the media source integration will not work, but all monitoring sensors will continue to function normally.

Author: with immich [1.136.0](https://github.com/immich-app/immich/releases/tag/v1.136.0) the [api key permission system](https://github.com/immich-app/immich/pull/20113) has been overhauled,  so this section needs a bit more love in a coming up PR

| **Disk used by photos** | Used disk space by photos (_admin only_) | ❌ |
| **Disk used by videos** | Used disk space by videos (_admin only_) | ❌ |

## Actions

### Upload file

This action allows you to upload a media file to your Immich instance. It takes the following arguments:

| Argument Name | Argument key | Required | Description |
| --- | --- | --- | --- |

> | Immich instance | `config_entry_id` | yes | Select the Immich instance where to upload the file. |

Author: ```suggestion
| Immich instance | `config_entry_id` | yes | The Immich instance where to upload the file. |
```

| File | `file` | yes | The path to the file to be uploaded.<br>_This path must be an entry in the `allowlist_external_dirs` in your [`homeassistant:`](/integrations/homeassistant/) section of your {% term "`configuration.yaml`" %} file._  |

> | Album id | `album_id` | no | The album where to put the file in after upload.<br>_To get the album id, open the Immich instance webui in a browser and navigate to the corresponding album, the album id can now be found in the URL `https://your-immich-instance/albums/<ALBUM-ID>`_ |

Author: ```suggestion
| Album ID | `album_id` | no | The album in which the file should be placed after uploading.<br>_To get the album id, open the Immich instance webui in a browser and navigate to the corresponding album, the album ID can now be found in the URL `https://your-immich-instance/albums/<ALBUM-ID>`_ |
```

An {% term update %} entity is created to inform about a new available Immich server version (_requires Immich server v1.134.0_).

## Actions

### Upload file

This action allows you to upload a media file to your Immich instance. It takes the following arguments:

> | Argument Name | Argument key | Required | Description |

Reviewer: we are trying to move away from tables if they get wide. Consider converting it into a list.

| --- | --- | --- | --- |
| Immich instance | `config_entry_id` | yes | The Immich instance where to upload the file. |
| File | `file` | yes | Use the [MediaSelector](/docs/blueprint/selectors/#media-selector) to define the file to be uploaded. |

> | Album ID | `album_id` | no | The album in which the file should be placed after uploading.<br>_To get the album id, open the Immich instance webui in a browser and navigate to the corresponding album, the album ID can now be found in the URL `https://your-immich-instance/albums/<ALBUM-ID>`_ |

Reviewer: ```suggestion
| Album ID | `album_id` | no | The album in which the file should be placed after uploading.<br>_To get the album ID, open the Immich instance web UI in a browser and navigate to the corresponding album, the album ID can now be found in the URL `https://your-immich-instance/albums/<ALBUM-ID>`_ |
```
Ignore this comment if I got it wrong. I'm assuming you're referring to the more generic term "web UI", and not the specific name [WebUI](https://webui.me/).
